### ISSUE TITLE
MaxwellSQLSyntaxError: 0b0000

### ISSUE THREAD

**Issue by XiaoyuBD** [zendesk/maxwell#1146]
Maxwell dead when parse a sql contains '`mortgage_item` BIT(4) NOT NULL DEFAULT 0b0000,'.

like:
CREATE TABLE `bug_test_bit` (
  `id` int(11) NOT NULL AUTO_INCREMENT COMMENT 'id',
  `operator` int(11) NOT NULL DEFAULT '0',
  `operator_value` varchar(20) NOT NULL DEFAULT '0',
  `created_at_ts` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at_ts` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `created_at_dt` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at_dt` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
 `mortgage_item` BIT(4) NOT NULL DEFAULT 0b0000,
  PRIMARY KEY (`id`))'

MAXWELL VERSION: 1.17.1
I tried other version (such as 1.13.*).It doesn't work neither.

log:

10:42:34,356 ERROR MysqlParserListener - (default_value DEFAULT 0b0000)
10:42:34,356 ERROR SchemaChange - Error parsing SQL: 'CREATE TABLE `bug_test_bit` (
  `id` int(11) NOT NULL AUTO_INCREMENT COMMENT 'id',
  `operator` int(11) NOT NULL DEFAULT '0' ,
  `operator_value` varchar(20) NOT NULL DEFAULT '0',
  `created_at_ts` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at_ts` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `created_at_dt` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at_dt` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
 `mortgage_item` BIT(4) NOT NULL DEFAULT 0b0000,
  PRIMARY KEY (`id`))'
10:42:34,356 ERROR AbstractSchemaStore - Error on bin log position Position[BinlogPosition[maxwell6-mysql-bin.000008:268143], lastHeartbeat=0]
com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: 0b0000
 at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:87)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
 at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:88)
 at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:100)
 at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:49)
 at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:99)
 at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:243)
 at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:260)
 at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:494)
 at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:150)
 at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:27)
 at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:224)
 at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156)
 at com.zendesk.maxwell.Maxwell.main(Maxwell.java:245)
10:42:34,358 INFO  BinlogConnectorLifecycleListener - Binlog disconnected.
10:42:34,359 INFO  TaskManager - Stopping 3 tasks

**Comment by XiaoyuBD** [comment]
I found that the exception occurs in  "mortgage_item BIT(4) NOT NULL DEFAULT 0b0000". "0b0000" could not be parsed correctly.

So I replace the "0b0000" with "b'0'" before parse the sql in com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(String currentDB, String sql) 

code is:
```
	private static List<SchemaChange> parseSQL(String currentDB, String sql) {
		if (sql.contains("BIT") || sql.contains("bit")) {
			sql = sql.replaceAll(" 0b0+", " b'0'");
			LOGGER.warn("sql replaced: " + sql);
		}
                other codes........

```

It works.
But I don't this it is good practice.

**Comment by osheroff** [comment]
fixed in 1.9.1
